libs/common/src/change-stream/delete-many.ts

```diff
@@ -50,11 +61,16 @@ const deleteManyWithAuditLogs = async ({ filter, collectionName, db, auditDetail
         throw new WriteConcernError();
       }
 
-      const deleteResult = await collection.deleteMany({ _id: { $in: documentsToDeleteIds } }, { session });
+      const deleteManyFilter = {
+        _id: { $in: documentsToDeleteIds },
+      } as unknown as Filter<WithoutId<DbModel<CollectionName>>>;
+      const deleteResult = await collection.deleteMany(deleteManyFilter, { session });
       if (!(deleteResult.acknowledged && deleteResult.deletedCount === documentsToDeleteIds.length)) {
         throw new WriteConcernError();
       }
+      transactionDeleteResult = { ...deleteResult };
```

Comment:
Because the mongo transaction doesn't return the value you would expect it to. My assumption was that it should probably return whatever the supplied function returns, but it doesn't, so we need to update the thing we want to return in the outer scope during the transaction